demos/blocklyfactory/block_exporter_tools.js

```diff
@@ -221,3 +210,61 @@ BlockExporterTools.prototype.generateToolboxFromLibrary
   xmlDom.appendChild(categoryElement);
   return xmlDom;
 };
+
+/**
+ * Generate xml for the workspace factory's category from imported block
+ * definitions.
+ *
+ * @return {!Element} Xml representation of a category.
+ */
+BlockExporterTools.prototype.generateCategoryFromBlockLib = function() {
+  // Create category DOM element.
+  var categoryElement = goog.dom.createDom('category');
+  categoryElement.setAttribute('name', 'Block Library');
+
+  var allBlockTypes = blockLibStorage.getBlockTypes();
+  // Object mapping block type to XML.
+  var blockXmlMap = blockLibStorage.getBlockXmlMap(allBlockTypes);
+
+  // Define the custom blocks in order to be able to create instances of
+  // them in the exporter workspace.
+  this.addBlockDefinitions(blockXmlMap);
+
+  for (var blockType in blockXmlMap) {
+    // Get block.
+    var block = FactoryUtils.getDefinedBlock(blockType, this.hiddenWorkspace);
+
+    // Get preview block XML.
+    var blockChild = Blockly.Xml.blockToDom(block);
```

Comment:
why do you call this child?  Looks to me like it is the xml of the block defined above.  Or is it the xml of the child of that block and I'm just misreading?